src/asset/asset.js

```diff
@@ -147,17 +156,33 @@ Object.assign(pc, function () {
             if (!this.file)
                 return null;
 
-            if (this.type === 'texture' || this.type === 'textureatlas') {
-                var device = this.registry._loader.getHandler('texture')._device;
-
-                if (this.variants.pvr && device.extCompressedTexturePVRTC) {
-                    return this.variants.pvr;
-                } else if (this.variants.dxt && device.extCompressedTextureS3TC) {
-                    return this.variants.dxt;
-                } else if (this.variants.etc2 && device.extCompressedTextureETC) {
-                    return this.variants.etc2;
-                } else if (this.variants.etc1 && device.extCompressedTextureETC1) {
-                    return this.variants.etc1;
+            if (this.type === 'texture' || this.type === 'textureatlas' || this.type === 'bundle') {
+                var app = this.registry._loader._app;
+                var device = app.graphicsDevice;
+
+                for (var i = 0, len = VARIANT_DEFAULT_PRIORITY.length; i < len; i++) {
+                    var variant = VARIANT_DEFAULT_PRIORITY[i];
+                    // if the device supports the variant
+                    if (! device[VARIANT_SUPPORT[variant]]) continue;
+
+                    // if the variant exists in the asset then just return it
+                    if (this.file.variants[variant]) {
+                        return this.file.variants[variant];
+                    }
+
+                    // if the variant does not exist but we the asset is in a bundle
+                    // and the bundle contain assets with this variant then return the default
+                    // file for the asset
+                    if (app._enableBundles) {
```

Comment:
This should be a public property if it's accessed outside the application

Reply:
Changed it to `enableBundles`